From 36bc9bfd9661f88039a8945688e07b0626b48610 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Tue, 15 Oct 2019 18:33:45 +0200
Subject: [PATCH 1/2] Fix Name conflict between plasma.desktop and
 plasmawayland.desktop

Summary:
SDDM until 0.18.1 appends " (Wayland)" to the name of any session file it
finds in wayland-sessions dir, and Plasma is relying on that behaviour to
distinguish between its X11 and Wayland sessions. This leads to duplicate
"Plasma" entries on any other DM not applying the same hack, e.g. lightdm,
and users are lost without downstream patching.

SDDM in 0.19 only appends " (Wayland)" in case the session name does
not already end with this, to avoid duplicating it.

CCBUG: https://bugs.kde.org/show_bug.cgi?id=368409
See also: https://github.com/sddm/sddm/pull/997

Reviewers: #plasma, davidedmundson, fvogt, ngraham

Reviewed By: #plasma, davidedmundson, fvogt, ngraham

Subscribers: ngraham, pino, rdieter, fvogt, davidedmundson, plasma-devel

Tags: #plasma

Differential Revision: https://phabricator.kde.org/D22210
---
 login-sessions/plasmawayland.desktop.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/login-sessions/plasmawayland.desktop.cmake b/login-sessions/plasmawayland.desktop.cmake
index 64f997ea4..3e9b46bc9 100644
--- a/login-sessions/plasmawayland.desktop.cmake
+++ b/login-sessions/plasmawayland.desktop.cmake
@@ -2,51 +2,7 @@
 Exec=dbus-run-session ${CMAKE_INSTALL_FULL_BINDIR}/startplasma-wayland
 TryExec=${CMAKE_INSTALL_FULL_BINDIR}/startplasma-wayland
 DesktopNames=KDE
-Name=Plasma
-Name[ar]=بلازما
-Name[bs]=Plazma
-Name[ca]=Plasma
-Name[ca@valencia]=Plasma
-Name[cs]=Plasma
-Name[da]=Plasma
-Name[de]=Plasma
-Name[el]=Plasma
-Name[en_GB]=Plasma
-Name[es]=Plasma
-Name[et]=Plasma
-Name[eu]=Plasma
-Name[fi]=Plasma
-Name[fr]=Plasma
-Name[gl]=Plasma
-Name[he]=פלזמה
-Name[hu]=Plasma
-Name[id]=Plasma
-Name[is]=Plasma
-Name[it]=Plasma
-Name[ja]=プラズマ
-Name[ko]=Plasma
-Name[lt]=Plasma
-Name[nb]=Plasma
-Name[nds]=Plasma
-Name[nl]=Plasma
-Name[nn]=Plasma
-Name[pa]=ਪਲਾਜ਼ਮਾ
-Name[pl]=Plazma
-Name[pt]=Plasma
-Name[pt_BR]=Plasma
-Name[ru]=Plasma
-Name[sk]=Plasma
-Name[sl]=Plasma
-Name[sr]=Плазма
-Name[sr@ijekavian]=Плазма
-Name[sr@ijekavianlatin]=Plasma
-Name[sr@latin]=Plasma
-Name[sv]=Plasma
-Name[tr]=Plama
-Name[uk]=Плазма
-Name[x-test]=xxPlasmaxx
-Name[zh_CN]=Plasma
-Name[zh_TW]=Plasma
+Name=Plasma (Wayland)
 Comment=Plasma by KDE
 Comment[ar]=بلازما كدي
 Comment[bs]=Plazma od strane KDe
-- 
2.23.0
